### DISCUSSION TITLE
Retrieve data from Daikin Altherma 3H HT ECH20

### DISCUSSION THREAD

**Discussion by Mychel60** [raomin/ESPAltherma#274] [General]
Hi Raomin, Hi all,

my Name is Michael and I am an 63 year old retired electric engineer and I am not so familiar with all the new possibilities on Internet. 

Since about 2 weeks I am owner of the Daikin Altherma **3H HT ECH20 (Type ETSXB16P50EF)** and the first time here on github.

Maybe you can help me out as I have following issue:

I have already IObroker and MQTT in place and I measure already the pump electric power consuption with a shelly 3EM - works fine with MQTT.
My idea is now to read out the generated "Heat Power" or the "Water Flow" in order to calculate the COP. Visualisation will be done via Influx and Grafana.

So far I followed the advice from my Daikin technician to use the **DAIKIN LAN Adapter BRP069A61** doing so.
But unfortunately, and this was confirmed by the Daikin Service Center, this will not work.
The results I got from this Adapter are very very poor.
In the same way the ONECTA App is also ridiculous, as I can just change 2 Temperatures and read out the electrical Power consumption.
In my opinion this is not state of the art cause all relevant data are persistent in the machine.

Anyway, I read the thread from Raomin concerning the AlthermaESP32 - this sounds extremely interesting to me. Great Job Raomin!
But checking the PCBs in the Pump I could not find the X10A interface connector on the main board which provides the RX/TX protocoll.
I just have 2 contacts (P1/P2) on X15 (12V) called "User Interface" (Those are ones where I connected the LAN Adapter).
I am not sure if I could use those connections for the ESP32

Does anyone have the same issue or can anyone advice me what else I could do?

Thanks a lot in advance for your feedback

Michael

**Comment by KalaNagHTD** [comment]
![9240A99D-C776-4FC6-B40A-57C912E08A1D](https://user-images.githubusercontent.com/77481230/234872216-4c5646f3-e232-4afc-805a-9de963ba1dee.jpeg)

![91A872C6-B013-44BC-A8B6-4FBF46D6B257](https://user-images.githubusercontent.com/77481230/234872045-cc82db50-3841-4726-96a1-355f189da0d1.jpeg)

**Comment by Mychel60** [comment]
Hello KalaNagHTD,

thank a lot for this information. Indeed, I found the X10A on the A1p board. Great stuff!!  Right after getting your information  I started the project with the AlthermaESP32 along Raomins instructions. Instead of HomeAssistant I am using Iobroker and it was  no problem to calculate the COP as I had already the MQTT Adapter installed. I wrote  2 short JS scripts for calculating "Q" and "COP":

**SCRIPT 2 - calculating "generated heat energy"**

`createState('javascript.0.WaermeMenge', 0, {unit: "W", type: "number", role: "value"});
var T4 = "javascript.0.Inlet temperature(R4T)"; //Vorlauftemperatur Verdichter
var T1 = "javascript.0.Outlet Water Heat Exch. Temp. (R1T)";// Rücklauftemperatur Verdichter
var Fl = "javascript.0.Flow sensor (l/min)"; // Durchflussmenge
//var Q = "WaermeMenge";

function  WaermeMenge() 
{
//Berechnung d. Wärmemenge
setState('WaermeMenge', getState(Fl).val * 1.163 * 60  * Math.abs(((273 + getState(T1).val) - (273 + getState(T4).val)))); 
}

on(Fl, WaermeMenge);
on(T4, WaermeMenge);
on(T1, WaermeMenge);
on (WaermeMenge, 

function ()
{
   WaermeMenge();
 });`

**SCRIPT 2: Calculation COP**

`createState('javascript.0.COP', 0, );

var P = "javascript.0.total_act"; //elektrische Leistung
var Q = "javascript.0.WaermeMenge"; // Wärmeleistung

function  COPkalk() 
{
//Berechnung d. COP
setState('COP', Math.abs(( 1000 * (getState(Q).val / getState(P).val))))
}
on(P, COPkalk);
on(Q, 
function ()
{
   COPkalk();
 });`

But obviously the calculated COP is wrong as the mean is around 6, in my opinion too high. Do you have an idea what could solve the issue?

Here is the final COP traced in grafana:
![
![image](https://user-images.githubusercontent.com/33604776/235440430-ad168f30-3979-4988-9854-7b5091e62c5c.png)
]

Here the Inlet and outlet temperature:
![
![image](https://user-images.githubusercontent.com/33604776/235441330-221e6695-034e-4d8e-8278-dbc22cf1478d.png)]

And finalle here the tracking of generated (and calculated) heatpower and the electric consumption (measured with Shelly 3EM)
![
![image](https://user-images.githubusercontent.com/33604776/235441650-006f4130-af4f-4592-bf58-27158844fc99.png)
]

Thank you very much for any comment.

**Comment by KalaNagHTD** [comment]
Your graphs show that the heat pump has very short heating cycles followed by longer pauses. Not good...

The heating power in your last chart does not follow the temperature graph or the electric power. On the contrary: the most heat generation is plotted when the electrical power is low and the temperatur difference is negative. That's impossible! So it's obvious that the heating power calculation is wrong.

**Reply by KalaNagHTD** [reply]
That's how my calculation of the thermal power looks like:
`schedule("*/5 * * * * *", function () {
  setStateDelayed("0_userdata.0.Thermische_Leistung"/*Thermische Leistung*/, Math.round((getState("0_userdata.0.Spreizung").val * getState("mqtt.0.rotex.flow_rate").val * 1.163)*10)/10, 100, false);
});
`

**Comment by Mychel60** [comment]
Thanks KalaNag  -  I have got the pump installed 4 weeks ago and I still waiting for the daikin technician to optimize it. The daikin Service seems to be fully overloaded currently. The cycle ratio of my pump is awful, -I know, but I could not find any parameters to change or improve it as I do not have the experience yet. I also dont know why Daikin does not provide the generated thermal Power via X10A interface cause I can check it on the HMI (Machine Display). 

I have incorporated the flow in l/min in the Graph. Does this look ok to you?
![image](https://user-images.githubusercontent.com/33604776/235490142-b0c5a5b7-62f2-4379-8014-497278b6ad6e.png)]

I also realized that in the cooling phase the outlet temperature is higher than the inlet temp. 
![image](https://user-images.githubusercontent.com/33604776/235486372-801a22f7-e3f3-4205-9e5c-9dfe87394b54.png)}

Then the Delta becomes negative, Therefore I used the "absolute" Math function in order to make the delta always positive. But I dont know if this is corrrect. So maybe I must not calulate the thermal power when the machine needs no electric power - in other words i should use just the heating cycle for calculation? The issue is that I do not get the "Spreizung" as a single value, I have to calculate it. 

Or there is a time stamp synchronisation issue - dont know but I will investigate.

So maybe I will wait for the Daikin technician first or do you have a further hintwhat I could do more?

**Comment by KalaNagHTD** [comment]
The problem that causes the frequent on/off cycles of your heat pump is caused by your house hydraulics most likely. The worst thing for a heat pump is using thermostatic valves when a buffer vessel is not installed.

The display shows the thermal energy, not the thermal power!

The flow rate received from X10A is measured in l/min. I get my flow rate via CAN bus in l/h.

I have no idea why Tout can be lower than Tin at the condenser heat exchanger! There is something wrong! Making an inverted temperature delta always positive doesn‘t solve the problem!

I‘d recommend discussing the heating system related questions in the HTD-Forum!

**Comment by Mychel60** [comment]
Thanks a lot for your interesting comments. I will discuss this with the Pump installer and with Daikin and will share the results on HTD.

**Reply by hattmann** [reply]
Hello Mychel60!

May I ask which espaltherma def file you use for the echo2 / ETSXB16P50E?
I have the same heatpump and i am currently not shure about the definition.
Thanks!

**Comment by Mychel60** [comment]
Hello,

thanks for asking.

I still have issues and I do not know what you mean with the def file.
Currently the project is suspended and I will restart in autumn.

What happened so far: My son supported me to retrieve the data from the
Heat-pump with ESP32. We got the data out of the pump in order to calculate
the generated heat power. Unfortunately the values are totally wrong
because the temp-delta becomes negative from time to time.

Currently I have not so much time, but as i mentioned before, I will
restart the project most likely with the HTD Forum.

Best regards

Mychel60


____________________________________________________________________________

Michael Richter
Dipl-Ing(FH)
St.-Gundekar-Str. 9
86650 Wemding
Mobil: +49-172 3576351
Privat: +49-9092-1442
Mail: ***@***.***
www: mr-electroservice.com


Am Fr., 11. Aug. 2023 um 19:01 Uhr schrieb hattmann <
***@***.***>:
